<a href="https://colab.research.google.com/github/Ashvik2008/Class114/blob/MASTER/Class114.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [96]:
import plotly.express as px
import pandas as pd
import csv
import numpy as np

In [97]:
df = pd.read_csv("/content/data.csv")

height=df["Height"].tolist()
weight=df["Weight"].tolist()

fig = px.scatter(x=height, y=weight)
fig.show()

As height increases, the weight also increases. this shows that height and weight are dependent on each other.

This line follows a formula, **y=mx+c** where 

y = weight, x = height, m = slope of the line and c = interdepth

In [98]:
#lets assume value of m as 1 and c as 0( following hit and trial method)
m = 1
c = 0
y = []

for x in height:
  y_value = m*x + c
  y.append(y_value)

#Plotting the points
fig = px.scatter(x=height, y=weight)
fig.update_layout(shapes=[
    dict(
      type= 'line',
      y0= min(y), y1= max(y),
      x0= min(height), x1= max(height)
    )
])
fig.show()

**Here, it looks like the line that we have taken is worlds apart. Let's try to change the values of M and C in such a way that this line fits well with the scatter plots.**

In [99]:
#lets assume value of m as 0.95 and c as -93( following hit and trial method)
m = 0.95
c = -93
y = []

for x in height:
  y_value = m*x + c
  y.append(y_value)

#Plotting the points
fig = px.scatter(x=height, y=weight)
fig.update_layout(shapes=[
    dict(
      type= 'line',
      y0= min(y), y1= max(y),
      x0= min(height), x1= max(height)
    )
])
fig.show()

Here, we have a close enough line that covers all the scatter plots in our chart. 
\
Here, the M is the slope of the line and the C is the intercept of the line.
\
Can we predict the weight of someone whose height is 250 using this formula? Let's see! **bold text**

In [100]:
x = 250
y = m * x + c

print(y)

144.5


Now we find values of m and c using computer slogorythms

In [101]:
import numpy as np

height_array = np.array(height)
weight_array = np.array(weight)

#Slope and intercept using pre-built function of Numpy
m, c = np.polyfit(height_array, weight_array, 1)


print(m)
print(c)
print("")

y = []
for x in height_array:
  y_value = m*x + c
  y.append(y_value)

#plotting the graph
fig = px.scatter(x=height_array, y=weight_array)
fig.update_layout(shapes=[
    dict(
      type= 'line',
      y0= min(y), y1= max(y),
      x0= min(height_array), x1= max(height_array)
    )
])
fig.show()

0.9600183750530678
-95.66248461433146



NOW WE CALCULATE WEIGHT OF PERSON WHOSE HEIGHT IS 250

In [102]:
df = pd.read_csv("/content/main.csv")

gre_Score=df["GRE Score"].tolist()
admit_Chance=df["Chance of Admit "].tolist()

gre_Score_array = np.array(gre_Score)
admit_Chance_array = np.array(admit_Chance)

#lets assume value of m as 0.95 and c as -93( following hit and trial method)
m = 0.009
c = -0.6
y = []

for x in height:
  y_value = m*x + c
  y.append(y_value)

#Plotting the points
fig = px.scatter(x=gre_Score, y=admit_Chance)
fig.update_layout(shapes=[
    dict(
      type= 'line',
      y0= min(y), y1= max(y),
      x0= min(gre_Score), x1= max(gre_Score)
    )
])
fig.show()

#Slope and intercept using pre-built function of Numpy
m, c = np.polyfit(gre_Score_array, admit_Chance_array, 1)

y = []
for x in gre_Score_array:
  y_value = m*x + c
  y.append(y_value)

#plotting the graph
fig = px.scatter(x=gre_Score_array, y=admit_Chance_array)
fig.update_layout(shapes=[
    dict(
      type= 'line',
      y0= min(y), y1= max(y),
      x0= min(gre_Score_array), x1= max(gre_Score_array)
    )
])
print("Using Formula")
fig.show()

print("")
print("Using Hit and trial method")

Using Formula



Using Hit and trial method


In [103]:
x = int(input("Enter GRE Score: "))

y = m*x+c
print(y)

Enter GRE Score: 337
0.925787997803571


If GRE Score is 337, Chance of admission is 0.92%